OpenRA.Game/TraitDictionary.cs

```diff
@@ -174,7 +175,8 @@ public T GetOrDefault(Actor actor)
 					return default(T);
 				else if (index + 1 < actors.Count && actors[index + 1] == actor)
 					throw new InvalidOperationException("Actor {0} has multiple traits of type `{1}`".F(actor.Info.Name, typeof(T)));
-				else return traits[index];
+				else
```

Comment:
You missed a few other style issues in here 🙂 

The elses are redundant and `default(T)` can now simplify to just `default`.